sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +702,116 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+// NOTE: For Nonfungible Assets
+// Byzantine Swaps: Timed counterpart for timed AssetLockListener for reversion
+const assetLockExpirationCallback = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hash: Hash,
+    expiryTime: number
+): void => {
+    // Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);
+    const reclaimCallback = async (contract: Contract, contractID: string) => {
+        // Check if asset hasn't been claimed yet. If true, do nothing. If false, either do the following cases:
+        const [islocked, isLockedQueryError] = await helpers.handlePromise(isAssetLockedInHTLCqueryUsingContractId(contract, contractID));
+        if (islocked == false) {
+            // CASE #1: Check GetHTLCHashPreImageByContractId(contractId)
+            const [result, evaluateError] = await helpers.handlePromise(
+                contract.evaluateTransaction("GetHTLCHashPreImageByContractId", contractID),
+            );
+            
+            // CASE #2: If the function above returns ANY error, call reclaimAssetInHTLC
+            if (evaluateError) {
+                // Put retry logic in event of failure. Retry 3x and then give up if unsuccessful (temp solution for now; requires CORE changes)
+                // If it fails, retry for i (arbitrarily defined) more attempts until you quit
+                let i = 0;
+                do {
+                    let [retryReclaimResult, retryReclaimableQueryError] = await helpers.handlePromise(reclaimAssetInHTLCusingContractId(contract, contractID));
+                    if (!retryReclaimableQueryError) {
+                        console.log("Nonfungible Asset unlocked successfully");
+                        break;
+                    }
+
+                    i++;
+                } while (i < 3);
+            }
+        }
+    }
+
+    // If you have time remaining, call setTimeout with reclaimCallback
+    if (expiryTime - currTimeSecs > 0) {
+        setTimeout(reclaimCallback, 1000 * (expiryTime - currTimeSecs), contract, contractID);
+    }
+
```

Comment:
Generally, it makes code less readable if you separate the `else` block with whitespace and comments. Just move `else if` to line 755 and append the comment to the end of that line.